packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -286,6 +295,13 @@ class DriveCommand extends RunCommandBase {
         androidEmulator: boolArgDeprecated('android-emulator'),
         profileMemory: stringArgDeprecated('profile-memory'),
       );
+      // If the test is sent a signal, take a screenshot before exiting
+      final Map<ProcessSignal, Object> screenshotTokens = _registerScreenshotCallbacks((ProcessSignal signal) async {
+        _logger!.printError('Caught $signal');
```

Comment:
I'll let you merge this and will rebase https://github.com/flutter/flutter/pull/114111 on top.